docs/architecture/adr-081-protobuf-mgmt.md

```diff
@@ -0,0 +1,202 @@
+# ADR 081: Protocol Buffers Management
+
+## Changelog
+
+- 2022-02-28: First draft
+
+## Status
+
+Accepted
+
+## Context
+
+At present, we manage the [Protocol Buffers] files ("protos") that define our
+wire-level data formats within the Tendermint repository itself (see the
+[`proto`](../../proto/) directory). Recently, we have been making use of [Buf],
+both locally and in CI, in order to generate Go stubs, and lint and check
+`.proto` files for breaking changes.
+
+The version of Buf used at the time of this decision was `v1beta1`, and it was
+discussed in [\#7975] and in weekly calls as to whether we should upgrade to
+`v1` and harmonize our approach with that used by the Cosmos SDK. The team
+managing the Cosmos SDK was primarily interested in having our protos versioned
+and easily accessible from the [Buf] registry.
+
+The three main customers for the `.proto` files and their needs, as currently
+understood, are as follows.
+
+1. Tendermint needs Go code generated from `.proto` files.
+2. Consumers of Tendermint's `.proto` files want to be able to access these
```

Comment:
Specifically, (other) tools that want to interoperate with Tendermint need to generate code for their own language. The `.proto` files themselves not really that interesting to consumers, except for as long as it takes to run the generator. Most projects seem to check in their generated code and only update it when they have to upgrade versions.

This might seem like a pointless distinction, but I think it's helpful to focus on the _outcome_ rather than the intermediate step.

Reply:
I think I've addressed this now.